cp23/hpi-fun2/whispers/executables/generator.cpp

```diff
@@ -0,0 +1,113 @@
+#include <testlib.h>
+#include <bits/stdc++.h>
+
+using namespace std;
+
+
+const string_view SAMPLE1 = R"(3 4
+0 1 1
+1 2 1
+2 3 2
+2 4 1)";
+
+const string_view SAMPLE2 = R"(2 4
+0 1 1
+2 1 1
+3 1 1
+4 3 1)";
+
+const string_view NO_EDGE = R"(2 1)";
+
+const string_view STAR = R"(1 8
+0 1 1
+2 1 1
+3 1 1
+4 1 1
+5 1 1
+6 1 1
+7 1 1)";
+
+const string_view WEIGHT_0 = R"(1 8
+0 1 0
+2 1 0
+3 1 0
+4 1 0
+5 1 0
+6 1 0
+7 1 0)";
+
+
+template <class F>
+void testcase(string name, string desc, F f) {
+    ofstream desc_file(name + ".desc");
+    desc_file << desc;
+    string in_file = name + ".in";
+    freopen(in_file.c_str(), "w", stdout);
+    f();
+}
+
+void predefined(string name, string desc, string_view content) {
+    testcase(name, desc, [&]() { cout << content; });
+}
+
+void sample(int num, string_view content) {
+    auto num_str = toString(num);
+    predefined("sample" + num_str, "Sample #" + num_str, content);
+}
+
+int main(int argc, char* argv[]) {
+    registerGen(argc, argv, 1);
+    rnd.setSeed(-2014268171001967338ll);
+
+    sample(1, SAMPLE1);
+    sample(2, SAMPLE2);
+    sample(3, NO_EDGE);
+    sample(4, STAR);
+    sample(5, WEIGHT_0);
+
```

Comment:
mit `sampls` generierte testcases können online eingesehen werden, ihr wollt wahrscheinlich das
```suggestion
    predefined("no_edge", "one node, no edge", NO_EDGE);
    predefined("star", "a star with 8 nodes", STAR);
    predefined("weight_0", "all edges have weight 0", WEIGHT_0);

```